### PR TITLE
Fix managed permission of analysis workflow for lab roles

### PR DESCRIPTION
**Pull request by ramonski**
## Description of the issue/feature this PR addresses

**NOTE: Please merge #2301 before this PR**

This PR changes the managed permission to view analysis results to all lab roles.

Furthermore, it allows the `Sampler` role to retract a field analysis result once it was submitted.

## Current behavior before PR

- Lab roles like e.g. `Sampler` or `Analyst` did not see analyses results in some states.
- Samplers were not allowed to retract a result once submitted in `Registered` state.

## Desired behavior after PR is merged

- All major lab roles can see analyses results in all states.
- Samplers can retract a result once submitted in `Registered` state.

--
I confirm I have tested this PR thoroughly and coded it according to [PEP8][1]
and [Plone's Python styleguide][2] standards.

[1]: https://www.python.org/dev/peps/pep-0008
[2]: https://docs.plone.org/develop/styleguide/python.html

### CODE REVIEW

File: src/senaite/core/upgrade/v02_05_000.py [hunk 1]
```diff
@@ -293,3 +295,43 @@ def _remove_action(type_info, action_id):
     index = actions.index(action_id)
     type_info.deleteActions([index])
     return _remove_action(type_info, action_id)
+
+
+def import_workflow(tool):
```
Comment (xispa) [inline src/senaite/core/upgrade/v02_05_000.py:301]: Please add the `@upgradestep(product, version)` decorator to `import_workflow` function. Workflows are modified by other add-ons quite often.

File: src/senaite/core/upgrade/v02_05_000.py [hunk 2]
```diff
@@ -293,3 +295,43 @@ def _remove_action(type_info, action_id):
...
+    """Helper to update role mappings for the given brains
+    """
+    wf_tool = api.get_tool("portal_workflow")
+    workflow = wf_tool.getWorkflowById(wf_id)
+    total = len(brains)
+    for num, brain in enumerate(brains):
+        if num and num % 100 == 0:
+            logger.info("Updating role mappings: {0}/{1}".format(num, total))
+        if num and num % 1000 == 0:
+            logger.info("Comitting {0}/{1}".format(num, total))
```
Comment (xispa) [inline src/senaite/core/upgrade/v02_05_000.py]: Typo: "comiting" -> "committing"

**Review by xispa — APPROVED** [review]
Two little things

File: src/senaite/core/upgrade/v02_05_000.py (hunk 1 above)
Comment (ramonski) [reply src/senaite/core/upgrade/v02_05_000.py:301]: Added in 5c35c7d1f

File: src/senaite/core/upgrade/v02_05_000.py (hunk 2 above)
Comment (ramonski) [reply src/senaite/core/upgrade/v02_05_000.py]: I'm always doing this wrong 🙈
Fixed in 6c1b69afa

**Review by xispa — APPROVED** [review]

**Pull request merged**